dpctl/tensor/libtensor/source/accumulators/cumulative_prod.cpp

```diff
@@ -171,7 +175,9 @@ struct CumProd1DIncludeInitialContigFactory
         if constexpr (TypePairSupportDataForProdAccumulation<srcTy,
                                                              dstTy>::is_defined)
         {
-            using ScanOpT = sycl::multiplies<dstTy>;
+            using ScanOpT = std::conditional_t<std::is_same_v<dstTy, bool>,
+                                               sycl::logical_and<dstTy>,
+                                               sycl::multiplies<dstTy>>;
```

Comment:
Also here is `IsSyclMultiplies` declared in `dpctl/tensor/libtensor/include/utils/sycl_utils.hpp`. Should it be reused? Or is that intended only for kernel code?

Reply:
It isn't obvious to me how it's useful here, for these conditional checks you mean?

Reply:
I misread the code, please ignore the comment.